lib/Corres_UL.thy

```diff
@@ -1536,4 +1541,92 @@ lemma corres_assert_opt_assume_l:
   \<Longrightarrow> corres_underlying sr nf nf' rrel (P and K (X \<noteq> None)) Q (assert_opt X >>= f) g"
   by (force simp: corres_underlying_def assert_opt_def return_def bind_def fail_def)
 
+section \<open> Preservation of state_relation for the concrete side symbolic execution \<close>
+
+(** sr_inv **)
+
+definition
+  sr_inv_ul :: "('a \<times> 'b) set \<Rightarrow> ('a \<Rightarrow> bool) \<Rightarrow> ('b \<Rightarrow> bool) \<Rightarrow> ('b \<Rightarrow> (unit \<times> 'b) set \<times> 'c) \<Rightarrow> bool" where
```

Comment:
It's not blocking, but I think it would be nice if there was a comment describing the intention behind this definition.